src/plugins/media/media.js

```diff
@@ -0,0 +1,261 @@
+        onStepleave,
+        onPlay,
+        onPause,
+        onEnded,
+        garbage;
+    
+    document.addEventListener("impress:init", function (event) {
+        root = event.target;
+        api = event.detail.api;
+        gc = api.lib.gc;
+        
+        if (root.dataset.mediaAutoplay === "" || root.dataset.mediaAutoplay === "true") {
+            root.mediaAutoplay = true;
+        } else {
+            root.mediaAutoplay = false;
+        }
+        if (root.dataset.mediaAutostop === "" || root.dataset.mediaAutostop === "true") {
+            root.mediaAutostop = true;
+        } else {
+            root.mediaAutostop = false;
+        }
+        if (root.dataset.mediaAutopause === "" || root.dataset.mediaAutopause === "true") {
+            root.mediaAutopause = true;
+        } else {
+            root.mediaAutopause = false;
+        }
+        
+        // this *must* be called only after setting the autopause, autoplay and autostop values
+        enhanceMedia();
+        
+        gc.pushCallback(garbage);
+    }, false);
+    
+    garbage = function () {
+        var elements, el, i;
+        removeMediaClasses();
+        delete root.mediaAutoplay;
+        delete root.mediaAutopause;
+        delete root.mediaAutopstop;
+        elements = document.querySelectorAll("#root, .step, audio, video");
+        for (i = 0; i < elements.length; i += 1) {
+            el = elements[i];
+            delete el.mediaAutoplay;
+            delete el.mediaAutopause;
+            delete el.mediaAutostop;
+        }
+        for (i = 0; i < gcAttributes.length; i += 1) {
+            el = gcAttributes[i];
+            el.node.removeAttribute(el.attr);
+        }
+        gcAttributes = [];
+    };
+    
+    onPlay = function (event) {
+        var type = event.target.nodeName.toLowerCase();
+        document.body.classList.add("impress-media-" + type + "-playing");
+        document.body.classList.remove("impress-media-" + type + "-paused");
+    };
+
+    onPause = function (event) {
+        var type = event.target.nodeName.toLowerCase();
+        document.body.classList.add("impress-media-" + type + "-paused");
+        document.body.classList.remove("impress-media-" + type + "-playing");
+    };
+
+    onEnded = function (event) {
+        var type = event.target.nodeName.toLowerCase();
+        document.body.classList.remove("impress-media-" + type + "-playing");
+        document.body.classList.remove("impress-media-" + type + "-paused");
+    };
+    
+    removeMediaClasses = function () {
+        var type, types;
+        types = ["video", "audio"];
+        for (type in types) {
+            document.body.classList.remove("impress-media-" + types[type] + "-playing");
+            document.body.classList.remove("impress-media-" + types[type] + "-paused");
+        }
+    };
+
+    enhanceMediaNodes = function () {
+        var i, id, media, mediaElement, type;
+        
+        //gc = event.detail.api.lib.gc;
```

Comment:
Above can be removed.

Reply:
Sorry, forgot to remove the comment. Seems to be an artifact of the development process.